dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava11Test.java

```diff
@@ -0,0 +1,126 @@
+import com.ibm.ws.jsp23.fat.JSPUtils;
+import com.meterware.httpunit.GetMethodWebRequest;
+import com.meterware.httpunit.WebConversation;
+import com.meterware.httpunit.WebRequest;
+import com.meterware.httpunit.WebResponse;
+
+import componenttest.annotation.MinimumJavaLevel;
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipForRepeat;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * JSP 2.3 tests which use Java 11 specific features.
+ *
+ * Tests must only run when Java 11 is in use. Runs only in FULL as Java 8 and 17 test also cover these options.
+ *
+ * Tests that just need to drive a simple request using our WebBrowser object can be placed in this class.
+ *
+ */
+// No need to run against cdi-2.0 since these tests don't use CDI at all.
+@Mode(TestMode.FULL)
+@MinimumJavaLevel(javaLevel = 11)
+@SkipForRepeat("CDI-2.0")
+@RunWith(FATRunner.class)
+public class JSPJava11Test {
+    private static final String APP_NAME = "TestJSPWithJava11";
+    private static final Logger LOG = Logger.getLogger(JSPJava11Test.class.getName());
+
+    @Server("jspJava11Server")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        ShrinkHelper.defaultDropinApp(server, APP_NAME + ".war");
+
+        server.startServer(JSPJava11Test.class.getSimpleName() + ".log");
+    }
+
+    @AfterClass
+    public static void testCleanup() throws Exception {
+        // Stop the server
+        if (server != null && server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    /**
+     * Simple test for index.jsp. Page uses Java 11's String#strip method
+     *
+     * @throws Exception
+     *                       if something goes horribly wrong
+     */
+    @Test
+    public void testJava11JSP() throws Exception {
+        WebConversation wc = new WebConversation();
+        wc.setExceptionsThrownOnErrorStatus(false);
+
+        String url = JSPUtils.createHttpUrlString(server, APP_NAME, "index.jsp");
+        LOG.info("url: " + url);
+
+        WebRequest request = new GetMethodWebRequest(url);
+        WebResponse response = wc.getResponse(request);
+        LOG.info("Servlet response : " + response.getText());
+
+        assertEquals("Expected " + 200 + " status code was not returned!",
+                     200, response.getResponseCode());
+        assertTrue("Java 11's String strip() method failed", response.getText().contains("success-strip"));
+        assertTrue("Java 11's String lines() did not return a stream.", response.getText().contains("java.util.stream.ReferencePipeline"));
+        assertTrue("Java 11's Map copyOf method failed", response.getText().contains("{key=success-copyof}"));
+    }
+
+    @Test
+    public void testJava11viaUseJDKCompiler() throws Exception {
```

Comment:
Let's add a javadoc so we know in the future how this differs from the other tests.